test/e2e/framework/util.go

```diff
@@ -3825,19 +3825,38 @@ func RestartKubeProxy(host string) error {
 
 func RestartKubelet(host string) error {
 	// TODO: Make it work for all providers and distros.
-	if !ProviderIs("gce", "aws") {
-		return fmt.Errorf("unsupported provider: %s", TestContext.Provider)
+	if !ProviderIs("gce", "aws", "vsphere") {
+		return fmt.Errorf("unsupported provider: %s, supported providers are: %s", TestContext.Provider, "gce, aws, vsphere")
```

Comment:
Having two different places for the same information means they can fall out of sync. Put them in a shared variable.